rust/nomad-base/src/agent.rs

```diff
@@ -152,4 +160,33 @@ pub trait NomadAgent: Send + Sync + std::fmt::Debug + AsRef<AgentCore> {
         })
         .instrument(span)
     }
+
+    /// Spawn a task which watches home for getting into failed state
+    #[allow(clippy::unit_arg)]
+    fn watch_home_fail(&self, interval: u64) -> Instrumented<JoinHandle<Result<()>>> {
```

Comment:
If you're taking the approach of returning a `FailedHome` error in the case that a failed home is detected (different than what the watcher does which is return `Result<State>)`, I would make sure the `watcher` also uses this functionality from `agent.rs` for consistency. The issues are different but its easiest to conceptualize if the watcher changes go in this PR too :)

Reply:
Basically, if you could choose a standardized approach for checking for home failures (e.g. if task returns `BaseError::FailedHome`) and use that everywhere, that'd be perfect

Reply:
See the comment above :)